### PR TITLE
Nc news7 

### PR DESCRIPTION
**Pull request by RTXVader**
atm there are quite a few console logs and probably not enough testing just wanted a review before lunch

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -157,41 +155,78 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  it("400: returns bad request when invalid ID", () => {
+    return request(app)
+      .post("/api/articles/'apple'/comments")
+      .send({ username: "butter_bridge", body: "apple" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
```
Comment (robert112233) [inline __tests__/app.test.js:252]: You are missing two tests here. 

404 - Username not found (you will _not_ need a check exists, psql will throw it's own error which can be caught with another if statement in your psql handler) 

400 - Missing keys, when you try to send {notusername: "butter_bridge", notbody: "valid text"} for example, again psql will throw an error for this, you shouldn't need promise.reject

File: modele/app.model.js [hunk 2]
```diff
@@ -88,3 +88,45 @@ exports.gettingCommentsByArticleId = (article_id) => {
...
+// console.log(query, 'query')
+  
+return db.query(articleQuery, [article_id]).then(({ rows }) => {
+  const articleExists = rows[0].exists;
+ 
+  if (!articleExists) {
+    return Promise.reject({
+      status: 404,
+      msg: "Not Found",
+    });
```
Comment (robert112233) [inline modele/app.model.js]: It shouldn't be necessary to check the article exists, psql will throw its own error, as it's every comment _must_ have an article_id which is referenced in the articles table

**Review by robert112233 — CHANGES_REQUESTED** [review]
Great work ! Only two missing tests for you to implement, add some console.logs in your controller's catch , and see what "err.code" is each time, then write some if statements using that err.code 

Please submit another PR for review

File: __tests__/app.test.js [hunk 3]
```diff
@@ -157,41 +155,96 @@ describe("GET /api/articles/:article_id/comments", () => {
       .expect(200)
       .then(({ body }) => {
         const { comments } = body;
-        comments.forEach((comment) => {
```
Comment (yalicee) [inline __tests__/app.test.js:160]: check length if you use a forEach

File: __tests__/app.test.js [hunk 4]
```diff
@@ -157,41 +155,96 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send({ username: "butter_bridge", body: "apple" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  it("404: returns NotFound when invalid username", () => {
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send({ username: ' ', body: "apple" })
```
Comment (yalicee) [inline __tests__/app.test.js]: please have a string with characters in so the test is for an invalid username and not an empty string

**Review by yalicee — APPROVED** [review]
Just noticed a few minor things but happy for you to merge after changes

**Pull request merged**